src/components/navigation-header/index.tsx

```diff
@@ -1,158 +1,223 @@
 import Text from 'components/text'
+
+// Local imports
 import { NavigationHeaderItem } from './types'
 import { NavigationHeaderDropdownMenu } from './components'
 import s from './navigation-header.module.css'
 
-const homePageNavItems = [
+/**
+ * Defines the navigation items for the main home page of the app. If this
+ * becomes authorable, it can be lifted into another area of the codebase.
+ */
+const HOME_PAGE_NAV_ITEMS = [
   { label: 'Documentation', pathSuffix: 'docs' },
   { label: 'Tutorials', pathSuffix: 'tutorials' },
   { label: 'Install', pathSuffix: 'downloads' },
 ]
 
-const productPageNavItems = [
+/**
+ * Defined the navigation items for all pages that live under `/{productSlug}`
+ * routes. If this becomes authorable, it can be lifted into another area of the
+ * codebase.
+ */
+const PRODUCT_PAGE_NAV_ITEMS = [
   { label: 'Home', pathSuffix: '' },
   { label: 'Documentation', id: 'documentation', isSubmenu: true },
   { label: 'Tutorials', pathSuffix: 'tutorials' },
   { label: 'Install', pathSuffix: 'downloads' },
 ]
 
-const NavigationHeader = () => {
+/**
+ * A mapping of Product slugs to their imported SVG colorwhite logos. Used for
+ * the headers under `/{productSlug}` pages.
+ */
+const PRODUCT_SLUGS_TO_LOGOS = {
+  vault: VaultLogo,
+  waypoint: WaypointLogo,
+}
+
+const HomePageHeaderContent = () => {
   const betaProductSlugs = __config.dev_dot.beta_product_slugs
+
+  return (
+    <div className={s.leftSide}>
+      <div className={s.contentBeforeNav}>
+        <InlineSvg
+          className={s.siteLogo}
+          src={require('./img/logo-white.svg?include')}
+        />
+      </div>
+      <nav className={s.nav}>
+        <ul className={s.navList}>
+          {HOME_PAGE_NAV_ITEMS.map(
+            (navItem: { label: string; pathSuffix: string }) => {
+              const { label, pathSuffix } = navItem
+              const menuItems = betaProductSlugs.map((slug: ProductSlug) => ({
+                icon: slug,
+                label: productSlugsToNames[slug],
+                path: `/${slug}/${pathSuffix}`,
+              }))
+              return (
+                <li key={pathSuffix}>
+                  <NavigationHeaderDropdownMenu
+                    itemGroups={[menuItems]}
+                    label={label}
+                  />
+                </li>
+              )
+            }
+          )}
+        </ul>
+      </nav>
+    </div>
+  )
+}
+
+const ProductPageHeaderContent = () => {
   const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
   const currentProduct = useCurrentProduct()
-
-  const Header = ({ children }: { children: ReactNode }) => (
-    <header className={s.root}>{children}</header>
+  const isBetaProduct = useIsBetaProduct(currentProduct.slug)
+  const betaProductSlugs = __config.dev_dot.beta_product_slugs
+  const companyLogo = (
+    <InlineSvg className={s.companyLogo} src={HashiCorpLogo} />
```

Comment:
Do we need this to be an inline svg or could we get away with it being an image?

Reply:
What are the tradeoffs?

Reply:
I would say if it's possible to render it as an image it would be preferable as it doesn't include the SVG in the bundle! If you need to adjust the SVG styles (fill/stroke/color/etc.) an image doesn't work though.

Reply:
@BRKalow the `height` and `width` of this svg are being adjusted in `src/components/navigation-header/navigation-header.module.css`, does that mean it's not possible to use an image?

Reply:
We can still style the `img` element, so we could control dimensions at that layer!